src/main/java/bio/terra/cli/businessobject/SpendProfileUser.java

```diff
@@ -0,0 +1,120 @@
+ * is not part of the current context or state.
+ */
+public class SpendProfileUser {
+  private static final Logger logger = LoggerFactory.getLogger(SpendProfileUser.class);
+
+  private String email;
+  private List<SpendProfilePolicy> policies;
+
+  private SpendProfileUser(String email, List<SpendProfilePolicy> policies) {
+    this.email = email;
+    this.policies = policies;
+  }
+
+  /**
+   * Enable a user on the default WSM spend profile.
+   *
+   * @param email email of the user to add
+   * @param policy policy to assign the user
+   */
+  public static SpendProfileUser enable(String email, SpendProfilePolicy policy) {
+    // call SAM to add a policy + email to the WSM default spend profile resource
+    new SpendProfileManagerService().enableUserForDefaultSpendProfile(policy, email);
+    logger.info("Enabled user on spend profile: email={}, policy={}", email, policy);
+
+    // return a SpendProfileUser = email + all policies (not just the one that was added here)
+    return getUser(email);
+  }
+
+  /**
+   * Disable a user on the default WSM spend profile.
+   *
+   * @param email email of the user to remove
+   * @param policy policy to remove from the user
+   */
+  public static SpendProfileUser disable(String email, SpendProfilePolicy policy) {
+    // call SAM to remove a policy + email from the WSM default spend profile resource
+    new SpendProfileManagerService().disableUserForDefaultSpendProfile(policy, email);
+    logger.info("Disabled user on spend profile: email={}, policy={}", email, policy);
+
+    // return a SpendProfileUser = email + all policies (not just the one that was removed here)
+    return getUser(email);
+  }
+
+  /** Get the spend profile user object. */
+  private static SpendProfileUser getUser(String email) {
+    // lowercase the email so there is a consistent way of looking up the email address
+    // the email address casing in SAM may not match the case of what is provided by the user
+    return listUsersInMap().get(email.toLowerCase());
+  }
+
+  /**
+   * List the users of the default WSM spend profile.
+   *
+   * @return a list of spend profile users
+   */
+  public static List<SpendProfileUser> list() {
+    return listUsersInMap().values().stream().collect(Collectors.toList());
+  }
+
+  /**
+   * Get the users of the default WSM spend profile in a map, to make it easy to lookup a particular
+   * user.
+   *
+   * @return a map of email -> spend profile user object
+   */
+  private static Map<String, SpendProfileUser> listUsersInMap() {
+    // call SAM to get the users + policies for the WSM default spend profile resource
+    List<AccessPolicyResponseEntry> accessPolicies =
+        new SpendProfileManagerService().listUsersOfDefaultSpendProfile();
+
+    // convert the SAM objects (policy -> list of emails) to CLI objects (email -> list of policies)
+    Map<String, SpendProfileUser> spendProfileUsers = new HashMap<>();
```

Comment:
can these maps be cached so you don't have to keep rebuilding them?

Reply:
I could cache them, but then we need a strategy for when to refresh them. Also, I think these commands to manage Sam groups and spend profile access are likely to be lightly-used commands, and so wouldn't benefit much from caching anyway.